### DISCUSSION TITLE
Feature suggestion: Remove the student user profile feature

### DISCUSSION THREAD

**Discussion by damithc** [TEAMMATES/teammates#11785] [Ideas]
Context: students who 'join' TEAMMATES using a Google account can create a user profile which includes a profile photo and the following details:
![image](https://user-images.githubusercontent.com/1673303/168438743-486da20b-2604-49d6-a5cc-1f3af59b8326.png)

I'm wondering if we should remove that feature. Some pros of removing:

1. The feature is not really core to TEAMMATES. Sharing information about students can be done by other means such as a regular LMS.
2. Requires Google login, which is considered 'optional' for students
3. Low usage (e.g., in April 2022, we added 17,000 new students but only 50+ created profiles i.e., about 0.3%, in March, it was 0.6%)
4. Removing it will allow us to remove the 'remind to join' feature as well; student profile is the only feature that incentivize instructors to make students join TEAMMATES. Without it, joining TEAMMATES becomes 'truly' optional for students. 
5. Other typical benefits: less code, less storage used
6. Lowers the chance of violating data privacy legislations as it reduces the amount of 'personal' data stored in TEAMMATES

Cons: the feature is implemented and working fine already.

What do others think?

**Comment by wkurniawan07** [comment]
As much as I am in favor towards deprecating unused/low-usage features (evidence is not lacking on how many of such feature deprecations are initiated by me in the past year), I have doubts on deprecating this.

> Sharing information about students can be done by other means such as a regular LMS.

I have not actually enquired on this matter for a long while. Which direction are we gearing TEAMMATES towards, as a product?

> Lowers the chance of violating data privacy legislations as it reduces the amount of 'personal' data stored in TEAMMATES

This is hardly a pro. Student profile itself is already an optional feature, so if the student chooses to enter his/her particulars, s/he does it on his/her own accord.

> Cons: the feature is implemented and working fine already.

This is not a con. No matter how well-implemented a feature is, if its value-add does not justify its maintenance burden, then it is very much a candidate for deprecation.

**Reply by damithc** [reply]
> I have not actually enquired on this matter for a long while. Which direction are we gearing TEAMMATES towards, as a product?

Good question. I've posted it as a separate discussion https://github.com/TEAMMATES/teammates/discussions/11789


> This is hardly a pro. Student profile itself is already an optional feature, so if the student chooses to enter his/her particulars, s/he does it on his/her own accord.

Yes, it being an optional feature is in our favor. Data privacy goes beyond that though. Once we have personal data stored in our system, the burden of compliance is with us irrespective of data submission was voluntary or not. For example, some regions require periodic audits by a third party. Plus, it increases risk of data theft, the damage if such a data were stolen, etc. That said, the data stored in the student profile is not that 'sensitive' and unlikely to make us a high-value target for data thieves.

> This is not a con. No matter how well-implemented a feature is, if its value-add does not justify its maintenance burden, then it is very much a candidate for deprecation.

Agreed.

**Comment by damithc** [comment]
Separately, another slightly 'odd' thing about the profiles feature is that it assumes the student wants to share the same data about self with instructors and students of _all_ courses.

> 4\. Removing it will allow us to remove the 'remind to join' feature as well; student profile is the only feature that incentivize instructors to make students join TEAMMATES. Without it, joining TEAMMATES becomes 'truly' optional for students.

This one has good value. Misguided use of 'remind to join' feature can cause support requests ("I joined but I can't see any sessions", "I don't have a Google account, what do I do?", ...). Furthermore, it requires lot of extra explanations to the instructor (as I realized while looking through help docs) as to when to use it, what benefits it brings, implications of students needing a Google account, etc. 
If we remove the 'join' feature entirely from the instructor's view, we can simplify their user experience quite a bit.

**Reply by wkurniawan07** [reply]
Actually, I read this entire paragraph as the downfall of us only supporting Google account. But that is precisely why Authentication Plus is a project.

**Comment by wkurniawan07** [comment]
If we are sure of removing this feature, let me elaborate on what exactly we can remove:
- Nationality-related and gender-related functions and validations.
- Every single picture-related functions, and the infra we needed to support them:
  - All pages/components with "hover to show photo". I don't think anyone will disagree that it is painful to implement such feature in a manner that caters to all browsers.
  - The API layer will be even simpler as all responses are now JSON.
  - We no longer need Google Cloud Storage. Although our GCS bill is non-existent, this removes a not small amount of code.

For some reason, I still need that little push to believe that it is safe to remove this feature.
